backends/p4tools/common/control_plane/symbolic_variables.h

```diff
@@ -0,0 +1,59 @@
+#ifndef BACKENDS_P4TOOLS_COMMON_CONTROL_PLANE_SYMBOLIC_VARIABLES_H_
+#define BACKENDS_P4TOOLS_COMMON_CONTROL_PLANE_SYMBOLIC_VARIABLES_H_
+
+#include "ir/ir.h"
+#include "ir/irutils.h"
+
+/// Defines accessors and utility functions for state that is managed by the control plane.
+/// This class can be extended by targets to customize initialization behavior and add
+/// target-specific utility functions.
+namespace P4Tools {
+
+namespace ControlPlaneState {
+
+/// @returns the symbolic boolean variable describing whether a table is configured by the
+/// control plane.
+const IR::SymbolicVariable *getTableActive(cstring tableName);
+
+/// @returns the symbolic variable describing a table match key.
+/// The table and field name are needed to generate a unique variable.
+const IR::SymbolicVariable *getTableKey(cstring tableName, cstring keyFieldName,
+                                        const IR::Type *type);
+
+/// @returns the symbolic variable describing a table ternary mask.
```

Comment:
I don't really understand what this is from the comment, can you elaborate a bit?